### PR TITLE
Add `--decimal` flag to benchmarking

### PR DESCRIPTION
**Pull request by austin362667**
## Summary

Improves follow-up of https://github.com/linkedin/Liger-Kernel/pull/196#pullrequestreview-2279028288.
To make decimal of speed benchmark configurable.

## Testing Done

- Hardware Type: NVIDIA L40
- [X] run `make test` to ensure correctness
- [X] run `make checkstyle` to ensure code style
- [X] run `make test-convergence` to ensure convergence
- [X] run `make run-benchmarks OVERWRITE=1` to ensure benchmarking successfully

### CODE REVIEW

**Review by ByronHsu — COMMENTED** [review]
this does not solve the issue because precision is configured on `do_bench`. triton now supports configurable precision for persisting csv: https://github.com/triton-lang/triton/blob/7480ef5028b724cb434b7841b016c6d6debf3b84/python/triton/testing.py#L295 (It was not before).  we can use that

also, we don't need to expose precision as an option because each kernel needs to use different precision. The way to check is to inspect the csv. If there are something like 0.9, 0.8, then we need to increase the prec.

tldr: use triton native argument, and manually tune the precision for the kernel if not precise enough

File: benchmark/data/all_benchmark_data.csv [hunk 1]
```diff
@@ -1,411 +1,411 @@
...
-swiglu,huggingface,full,memory,MB,T,sequence length,1024,1294.25,1294.25,1294.25,"{""B"": 4, ""hidden_size"": 4096, ""dtype"": ""torch.bfloat16"", ""intermediate_size"": 11008, ""hidden_act"": ""silu""}",NVIDIA A100-SXM4-80GB,2024-09-03 15:40:36,0.2.1
-swiglu,huggingface,full,memory,MB,T,sequence length,2048,1992.25,1992.25,1992.25,"{""B"": 4, ""hidden_size"": 4096, ""dtype"": ""torch.bfloat16"", ""intermediate_size"": 11008, ""hidden_act"": ""silu""}",NVIDIA A100-SXM4-80GB,2024-09-03 15:40:36,0.2.1
-swiglu,huggingface,full,memory,MB,T,sequence length,4096,3452.25,3452.25,3452.25,"{""B"": 4, ""hidden_size"": 4096, ""dtype"": ""torch.bfloat16"", ""intermediate_size"": 11008, ""hidden_act"": ""silu""}",NVIDIA A100-SXM4-80GB,2024-09-03 15:40:36,0.2.1
-swiglu,huggingface,full,memory,MB,T,sequence length,8192,6372.25,6372.25,6372.25,"{""B"": 4, ""hidden_size"": 4096, ""dtype"": ""torch.bfloat16"", ""intermediate_size"": 11008, ""hidden_act"": ""silu""}",NVIDIA A100-SXM4-80GB,2024-09-03 15:40:36,0.2.1
+layer_norm,liger,forward,speed,ms,N,hidden size,1024,0.0573,0.0563,0.0594,"{""M"": 4096, ""dtype"": ""torch.float32"", ""eps"": 1e-06}",NVIDIA L40S,2024-09-04 16:00:49,0.2.1
```
Comment (yundai424) [inline benchmark/data/all_benchmark_data.csv:2]: maybe we can keep the bench result on A100 -- or better off, maintain separate csv data for each hardware? 🤔 @shimizust @ByronHsu wdyt

Comment (ByronHsu) [reply benchmark/data/all_benchmark_data.csv:2]: yes was asking austin to rerun on A100

**Comment by ByronHsu** [comment]
https://github.com/linkedin/Liger-Kernel/pull/196#issuecomment-2329629032 i think no action is needed for now

**Comment by austin362667** [comment]
The PR https://github.com/linkedin/Liger-Kernel/pull/196/files#diff-321a032a217a6c93e14d1cae93dd20922d020e0d55893220a026ac5c52374d81R151 already solved the under precision issue with its own csv writer, no action is needed for now, closing~

File: benchmark/data/all_benchmark_data.csv (hunk 1 above)
Comment (shimizust) [reply benchmark/data/all_benchmark_data.csv:2]: I'm not sure if the original file was deleted and re-run, but the behavior of `make run-benchmarks` or `make run-benchmarks OVERWRITE=1` in this case should have been to append entries to the existing CSV file since it was using different hardware spec. The `OVERWRITE=1` flag just means if test was run using the same parameters (including hardware), it will overwrite the existing record with the latest data.

This is essentially a file-based DB, so I don't think there's a need to split into different files. One can easily filter by GPU name when loading this data for analysis. Building up benchmarks on different GPU here could be useful.
